### PR TITLE
move sponge from constraint system to srs

### PR DESCRIPTION
**Pull request by fabrizio-m**
Should solve #591

### CODE REVIEW

File: circuit-construction/tests/example_proof.rs [hunk 1]
```diff
@@ -64,23 +64,20 @@ const PUBLIC_INPUT_LENGTH: usize = 3;
 #[test]
 fn test_example_circuit() {
     // create SRS
+    let proof_system_constants = fp_constants();
     let srs = {
-        let mut srs = SRS::<Affine>::create(1 << 7); // 2^7 = 128
+        let mut srs = SRS::<Affine>::create(1 << 7, proof_system_constants.poseidon.clone()); // 2^7 = 128
```
Comment (mimoo) [inline circuit-construction/tests/example_proof.rs:69]: I'm wondering if we really have to pass that as argument. In the same vein we generate `endo_q` and `endo_r`, couldn't we generate this automatically in `SRS::create`?

File: kimchi/src/prover_index.rs [hunk 2]
```diff
@@ -93,6 +94,19 @@ where
             fq_sponge_params,
         }
     }
+
+    /// This function verifies the consistency of the wire
+    /// assignements (witness) against the constraints
+    ///     witness: wire assignement witness
+    ///     RETURN: verification status
+    pub fn verify(
```
Comment (mimoo) [inline kimchi/src/prover_index.rs:102]: maybe `verify_witness`?

File: kimchi/src/verifier_index.rs [hunk 3]
```diff
@@ -238,7 +238,7 @@ where
             endo: self.cs.endo,
             lookup_index,
             linearization: self.linearization.clone(),
-            fr_sponge_params: self.cs.fr_sponge_params.clone(),
+            fr_sponge_params: self.srs.scalar_sponge_params.clone(),
             fq_sponge_params: self.fq_sponge_params.clone(),
```
Comment (mimoo) [inline kimchi/src/verifier_index.rs:242]: I guess `fq_sponge_params` could follow the same treatment right? The SRS could contain both parameters

File: kimchi/src/verifier_index.rs [hunk 4]
```diff
@@ -238,7 +238,7 @@ where
             endo: self.cs.endo,
             lookup_index,
             linearization: self.linearization.clone(),
-            fr_sponge_params: self.cs.fr_sponge_params.clone(),
+            fr_sponge_params: self.srs.scalar_sponge_params.clone(),
```
Comment (mimoo) [inline kimchi/src/verifier_index.rs:241]: if this is present in the SRS, then we can remove it from the verifier index unless I'm missing something

File: kimchi/src/prover_index.rs (hunk 2 above)
Comment (fabrizio-m) [reply kimchi/src/prover_index.rs:102]: Sounds good.

File: kimchi/src/verifier_index.rs (hunk 4 above)
Comment (fabrizio-m) [reply kimchi/src/verifier_index.rs:241]: The problem with that is VerifierIndex::srs() that returns the srs or tries to create it, in that case VerifierIndex won't be able to provide the params to SRS::create(). VerifierIndex::srs() could take the params but I think that will just be pushing the problem.
But my proposal about using a trait should solve this too.

**Comment by fabrizio-m** [comment]
Closing in favor of the more general #639 that will address what this tried to solve in a future PR.

**Pull request closed without merging**